pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1018,46 +1192,46 @@ def __init__(
         ):
             raise ValueError("The work wires must be different from the control and target wires")
 
-        self._target_wire = wires[0]
-        self._work_wires = work_wires
-        self._control_wires = control_wires
-
-        wires = control_wires + wires
-
-        if not control_values:
-            control_values = "1" * len(control_wires)
-
-        control_int = self._parse_control_values(control_wires, control_values)
-        self.control_values = control_values
+        self._hyperparameters = {"target_wire": wires[0],
+                                 "work_wires": work_wires,
+                                 "control_wires": control_wires,
+                                 "control_values": control_values}
 
-        self._padding_left = control_int * 2
-        self._padding_right = 2 ** len(wires) - 2 - self._padding_left
-        self._CX = None
+        total_wires = control_wires + wires
 
-        super().__init__(*params, wires=wires, do_queue=do_queue)
+        super().__init__(wires=total_wires, do_queue=do_queue)
 
     @property
     def num_params(self):
         return 0
 
-    def _matrix(self, *params):
-        if self._CX is None:
-            self._CX = block_diag(
-                np.eye(self._padding_left), PauliX.matrix, np.eye(self._padding_right)
-            )
+    @staticmethod
+    def compute_matrix(control_wires, control_values=None):
+        """Canonical matrix representation of the MultiControlledX operator.
 
-        return self._CX
+        Args:
+           control_wires (Iterable): wires to place controls on
+           control_values (str): string of bits determining the controls
```

Comment:
I changed it to fit the usual pattern. These are part of the hyperparameters of `MultiControlledX`. It has no trainable params.